packages/atlas/src/providers/transactionManager/useTransaction.ts

```diff
@@ -100,26 +108,58 @@ export const useTransaction = (): HandleTransactionFn => {
           addBlockAction({ callback: syncCallback, targetBlock: result.block })
         })
 
-        return new Promise((resolve) => {
-          queryNodeSyncPromise.then(() => {
-            if (!minimized) {
-              setDialogStep(ExtrinsicStatus.Completed)
+        await queryNodeSyncPromise
+
+        if (result.transactionHash) {
+          const { data } = await getTransactionStatus({ variables: { transactionHash: result.transactionHash } })
+          const status = data?.metaprotocolTransactionStatusEvents[0]?.status
+
+          if (!status) {
+            for (let i = 0; i <= RETRIES; i++) {
+              ConsoleLogger.warn(`No transcation status event found - retries: ${i}/${RETRIES}`)
+              await wait(RETRIES_TIMEOUT)
+              const { data } = await refetch()
+              const status = data?.metaprotocolTransactionStatusEvents[0]?.status
+              if (status) {
+                break
+              }
+              if (i === 10 && !status) {
+                throw new JoystreamLibError({
+                  name: 'MetaprotocolTransactionError',
+                  message: 'No transcation status event found',
+                  details: result,
+                })
+              }
             }
-            snackbarSuccessMessage &&
-              displaySnackbar({
-                ...snackbarSuccessMessage,
-                iconType: 'success',
-                timeout: TX_COMPLETED_SNACKBAR_TIMEOUT,
-              })
-            resolve(true)
+          }
+
+          if (status?.__typename === 'MetaprotocolTransactionErrored') {
```

Comment:
This will not work properly if we got `status` during retries - in your `for` loop, `status` is declared as scoped variable so when you check `status` in this line you are accessing the one from L115, even if we got a response during retry